File: pkg/dev_compiler/test/codegen_expected/lib/html/js_typed_interop_anonymous_exp_test.js
dart_library.library('lib/html/js_typed_interop_anonymous_exp_test', null, /* Imports */[
  'dart_sdk',
  'unittest'
], function load__js_typed_interop_anonymous_exp_test(exports, dart_sdk, unittest) {
  'use strict';
  const core = dart_sdk.core;
  const dart = dart_sdk.dart;
  const dartx = dart_sdk.dartx;
  const html_config = unittest.html_config;
  const unittest$ = unittest.unittest;
  const src__matcher__expect = unittest.src__matcher__expect;
  const src__matcher__core_matchers = unittest.src__matcher__core_matchers;
  const js_typed_interop_anonymous_exp_test = Object.create(null);
  let VoidTodynamic = () => (VoidTodynamic = dart.constFn(dart.definiteFunctionType(dart.dynamic, [])))();
  js_typed_interop_anonymous_exp_test.main = function() {
    html_config.useHtmlConfiguration();
    unittest$.test('simple', dart.fn(() => {
      let l = {x: 3, y: "foo"};
      src__matcher__expect.expect(l.x, src__matcher__core_matchers.equals(3));
      src__matcher__expect.expect(l.y, src__matcher__core_matchers.equals("foo"));
      src__matcher__expect.expect(l.z, src__matcher__core_matchers.isNull);
    }, VoidTodynamic()));
  };
  dart.fn(js_typed_interop_anonymous_exp_test.main, VoidTodynamic());
  // Exports:
  exports.js_typed_interop_anonymous_exp_test = js_typed_interop_anonymous_exp_test;
});

File: tests/html/js_typed_interop_anonymous_exp_test.dart
// Copyright (c) 2015, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.
//
// SharedOptions=--experimental-trust-js-interop-type-annotations

// Same test as js_typed_interop_anonymous, but using the
// --experimental-trust-js-interop-type-annotations flag.
@JS()
library js_typed_interop_anonymous_exp_test;

import 'dart:html';
import 'dart:js' as js;

import 'package:js/js.dart';
import 'package:unittest/unittest.dart';
import 'package:unittest/html_config.dart';

@JS()
@anonymous
class Literal {
  external factory Literal({int x, String y, num z});

  external int get x;
  external String get y;
  external num get z;
}

main() {
  useHtmlConfiguration();

  test('simple', () {
    var l = new Literal(x: 3, y: "foo");
    expect(l.x, equals(3));
    expect(l.y, equals("foo"));
    expect(l.z, isNull);
  });
}
